tests/python/pants_test/engine/legacy/test_graph_integration.py

```diff
@@ -0,0 +1,178 @@
+    })
+    self.assert_success(pants_run)
+
+    for as_zsh_glob in expected_globs:
+      warning_msg = self._ERR_FMT.format(
+        base=self._SOURCES_TARGET_BASE,
+        name=target_name,
+        desc='sources',
+        glob=glob_str,
+        as_zsh_glob=as_zsh_glob)
+      self.assertIn(warning_msg, pants_run.stderr_data)
+
+  @unittest.skip('Skipped to expedite landing #5769: see #5863')
+  def test_missing_sources_warnings(self):
+    for target_name in self._SOURCES_ERR_MSGS.keys():
+      self._list_target_check_warnings(target_name)
+
+  @unittest.skip('Skipped to expedite landing #5769: see #5863')
+  def test_existing_sources(self):
+    target_full = '{}:text'.format(self._SOURCES_TARGET_BASE)
+    pants_run = self.run_pants(['list', target_full], config={
+      GLOBAL_SCOPE_CONFIG_SECTION: {
+        'glob_expansion_failure': 'warn',
+      },
+    })
+    self.assert_success(pants_run)
+    self.assertNotIn("WARN]", pants_run.stderr_data)
+
+  @unittest.skip('Skipped to expedite landing #5769: see #5863')
+  def test_missing_bundles_warnings(self):
+    target_full = '{}:{}'.format(self._BUNDLE_TARGET_BASE, self._BUNDLE_TARGET_NAME)
+    pants_run = self.run_pants(['list', target_full], config={
+      GLOBAL_SCOPE_CONFIG_SECTION: {
+        'glob_expansion_failure': 'warn',
+      },
+    })
+
+    self.assert_success(pants_run)
+
+    # FIXME: this fails because we just report the string arguments passed to rglobs(), not the
+    # zsh-style globs that they expand to (as we did in the previous iteration of this). We should
+    # provide both the specific argument that failed, as well as the glob it expanded to for
+    # clarity.
+    for glob_str, expected_globs in self._BUNDLE_ERR_MSGS:
+      for as_zsh_glob in expected_globs:
+        warning_msg = self._ERR_FMT.format(
+          base=self._BUNDLE_TARGET_BASE,
+          name=self._BUNDLE_TARGET_NAME,
+          desc='fileset',
+          glob=glob_str,
+          as_zsh_glob=as_zsh_glob)
+        self.assertIn(warning_msg, pants_run.stderr_data)
+
+  @unittest.skip('Skipped to expedite landing #5769: see #5863')
+  def test_existing_bundles(self):
+    target_full = '{}:mapper'.format(self._BUNDLE_TARGET_BASE)
+    pants_run = self.run_pants(['list', target_full], config={
+      GLOBAL_SCOPE_CONFIG_SECTION: {
+        'glob_expansion_failure': 'warn',
+      },
+    })
+    self.assert_success(pants_run)
+    self.assertNotIn("WARN]", pants_run.stderr_data)
+
+  @unittest.skip('Skipped to expedite landing #5769: see #5863')
+  def test_exception_with_global_option(self):
+    sources_target_full = '{}:some-missing-some-not'.format(self._SOURCES_TARGET_BASE)
+
+    pants_run = self.run_pants(['list', sources_target_full], config={
+      GLOBAL_SCOPE_CONFIG_SECTION: {
+        'glob_expansion_failure': 'error',
+      },
+    })
+    self.assert_failure(pants_run)
+    self.assertIn(AddressLookupError.__name__, pants_run.stderr_data)
+    expected_msg = """
```

Comment:
This is much, much too specific. Changing either error reporting or the shape of the rule graph would break it.

Should validate a much more specific thing... maybe just that both the target address and glob are included.

Reply:
c3750a494f6569e6f72876b66ee60a1d117103ce should make this much more useful.